pkg/operator/mon/mon.go

```diff
@@ -376,18 +301,30 @@ func (c *Cluster) waitForPodToStart(name string) (string, error) {
 
 	// Poll the status of the pods to see if they are ready
 	status := ""
-	for i := 0; i < c.context.MaxRetries; i++ {
+	retryFactor := 3
+	for i := 0; i < c.context.MaxRetries*retryFactor; i++ {
 		// wait and try again
-		logger.Infof("waiting %ds for pod %s to start. status=%s", c.context.RetryDelay, name, status)
-		<-time.After(time.Duration(c.context.RetryDelay) * time.Second)
+		if i%retryFactor == 0 {
+			logger.Infof("waiting for mon %s to start. status=%s", name, status)
+		}
+		<-time.After(time.Duration(c.context.RetryDelay/retryFactor) * time.Second)
 
-		pod, err := c.context.Clientset.CoreV1().Pods(c.Namespace).Get(name, metav1.GetOptions{})
+		options := metav1.ListOptions{LabelSelector: fmt.Sprintf("mon=%s", name)}
+		pods, err := c.context.Clientset.CoreV1().Pods(c.Namespace).List(options)
 		if err != nil {
-			return "", fmt.Errorf("failed to get mon pod %s. %+v", name, err)
+			return "", fmt.Errorf("failed to get mon pod. %+v", err)
+		}
+		if len(pods.Items) == 0 {
+			logger.Infof("%s pod not yet created", name)
+			continue
+		}
+		if len(pods.Items) > 1 {
+			logger.Warningf("more than one mon pod found for %s", name)
```

Comment:
if there's more than 1 mon pod, is there something we need to do to rectify that situation?